autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/UpdateFailedHandler.java

```diff
@@ -51,15 +58,30 @@ public void onApplicationEvent(UpdateFailedEvent event) {
         if (failed < RETRY_THRESHOLD) {
             updateFailures.put(autoscaleClusterId, failed);
             LOGGER.debug("Increased failed count '{}' for cluster '{}'", failed, cluster.getStackCrn());
+
+            if (event.getCausedBy() != null && (event.getCausedBy() instanceof ForbiddenException)) {
+                Integer forbiddencount = getForbiddenFailureCount(autoscaleClusterId) + 1;
+                forbiddenFailures.put(autoscaleClusterId, forbiddencount);
+                LOGGER.debug("Increased forbidden count '{}' for cluster '{}'", forbiddencount, cluster.getStackCrn());
+            }
         } else {
+            if (getForbiddenFailureCount(autoscaleClusterId) >= (RETRY_THRESHOLD - 1)) {
```

Comment:
Not sure how feasible this. Would be worth evaluating if we can fallback to the regular user if there's an AuthN failure with the machine user, or vice-versa. Essentially - if a sync does not complete, don't break autoscale for clusters where it is already running properly.

Reply:
Or maybe there's an IPA API which can tell us whether the sync to IPA is complete or not?

Reply:
I think there is a separate API call to check the completion of userSync as @horadla23 mentioned
>the API calls to FMS will not wait until syncs are finished, just will initialize them afaik
so you probably need to check with different call if sync is finished